src/toil/batchSystems/slurm.py

```diff
@@ -467,6 +496,24 @@ def parse_elapsed(self, elapsed):
                 pass  # slurm may return INVALID instead of a time
             return total_seconds
 
+    def __init__(self, config, maxCores, maxMemory, maxDisk):
+        super().__init__(config, maxCores, maxMemory, maxDisk)
+        self._get_partition_info()
+
+    def _get_partition_info(self):
+        sinfo_command = ["sinfo",
+                         "-a",
+                         "-o",
+                         "%P %G %l %p %c %m"]
+
+        sinfo = call_command(sinfo_command)
+
+        self.partitions = []
+        for line in sinfo.split("\n")[1:]:
+            if line.strip():
+                partition_name, gres, time, priority, cpus, memory = line.split(" ")
+                self.partitions.append((partition_name.rstrip("*"), gres != "(null)", time, priority, cpus, memory))
```

Comment:
Yeah, this is a complex enough tuple that it probably should be a `NamedTuple` type with field names.